acceptance/launcher_test.go

```diff
@@ -0,0 +1,215 @@
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "Executing web process-type"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute web:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+	})
+
+	when("no args and CNB_PROCESS_TYPE", func() {
+		it("launches given default process", func() {
+			cmd := exec.Command("docker", "run", "--rm", "--env", "CNB_PROCESS_TYPE=other-process", launchImage)
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "Executing other-process process-type"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute other-process:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+	})
+
+	when("process provided in CMD", func() {
+		it("launches process", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage, "other-process")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "Executing other-process process-type"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute other-process:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+	})
+
+	when("CMD provided", func() {
+		it("sources profiles and launches command with shell", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage, "echo my-cmd")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "sourced bp profile\nsourced app profile\nmy-cmd"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute provided CMD:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+
+		it("sets env vars from layers", func() {
+			cmd := exec.Command("docker", "run", "--rm", launchImage, "echo $SOME_VAR $OTHER_VAR")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			expected := "sourced bp profile\nsourced app profile\nsome-bp-val other-bp-val"
+			if !strings.Contains(string(output), expected) {
+				t.Fatalf("failed to execute provided CMD:\n\t got: %s\n\t want: %s", output, expected)
+			}
+		})
+
+		it("passes through env vars from user, excluding blacklist", func() {
+			cmd := exec.Command("docker", "run", "--rm",
+				"-eCNB_APP_DIR=/workspace",
+				"-eSOME_USER_VAR=some-user-val",
+				"-eOTHER_VAR=other-user-val",
+				launchImage,
+				"echo $SOME_USER_VAR $CNB_APP_DIR $OTHER_VAR")
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("failed to run %v\n OUTPUT: %s\n ERROR: %s\n", cmd.Args, output, err)
+			}
+			// bp appends other-bp-val with delimeter '**'
```

Comment:
Perhaps it's worth adding a separate test for this behavior? Or would that belong somewhere else?

Reply:
I was trying to strike a balance between small targeted tests and speed (not starting too many containers). Happy to break it out though if it makes this test confusing.

Reply:
That makes sense to me. I think we can leave it where it is for now...